[Jiayan Ben] iP

J-Y-Yan opened the pull request.

J-Y-Yan committed:
add file "Hello.txt"

J-Y-Yan committed:
add new file "HelloAgain.txt"

J-Y-Yan committed:
remove the file "Hello.txt" and "HelloAgain.txt"

J-Y-Yan committed:
add new file "Deadline.java" as subclass of Tast

J-Y-Yan committed:
implement the main function Duke to finish the required tasks from lv1 - 4

J-Y-Yan committed:
add new file "Event.java" as subclass of Task

J-Y-Yan committed:
add new file "Task.java" as parent class of task objects

J-Y-Yan committed:
add new file "Todo.java" as subclass of Task

J-Y-Yan committed:
add file "EXPECTED.txt" to test the output. ***content not yet initialised

J-Y-Yan committed:
add "input.txt" containing the input commands

J-Y-Yan committed:
add "runtest.bat" to compile the file. ***main class location is not yet mateched

StanleyW00 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,97 @@
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        Scanner scanner = new Scanner(System.in);
+        Task[] list = new Task[100];
+        int count = 0;
```
Perhaps the variable names should be renamed specifically to task in order to avoid confusion in the future.

```suggestion
        Task[] tasks = new Task[100];
        int tasksCount = 0;
```

StanleyW00 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,27 @@
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    public void Done(){
+        this.isDone = true;
+    }
+
+    public void unDone(){
+        this.isDone = false;
+    }
```
Perhaps these methods should be renamed to represent verbs.

```suggestion
    public void markAsDone(){
        this.isDone = true;
    }

    public void unmarkAsDone(){
        this.isDone = false;
    }
```

StanleyW00 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,97 @@
+
+            case "unmark":
+                int task = Integer.parseInt(parts[1]);   //assume  saved in part[1], need further improve
+                if( (task > count ) || (task <1) ){
+                    System.out.println("Oops! You don't have the task in this positions.");
+                }
```
I like that you checked whether the task position is there or not.

StanleyW00 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,97 @@
+                break;
+
+            case "todo":
+                // command e.g. todo borrow book
```
I like that you put the example inputs for each type of tasks.

StanleyW00 reviewed (commented):
Looks nice.
Just need to fix some of the names.

Cheezeblokz commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,14 @@
+public class Deadline extends Task {
+
+    protected String by;
```
Confusing variable name, use noun for variables.
```suggestion
    protected String userName;
```

J-Y-Yan replied:
Thanks for your suggestions. I have improved the naming for a few variables to avoid confusion.

Cheezeblokz commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,97 @@
+        System.out.println("What can I do for you?");
+
+        while (true) {
+            String command = scanner.nextLine();
+            String[] parts = command.split(" ");
```
Unspecific variable naming.
```suggestion
            String userCommand = scanner.nextLine();
            String[] userInputParts = userCommand.split(" ");
```

Cheezeblokz commented on src/main/java/Todo.java:
```diff
@@ -0,0 +1,11 @@
+    }
+
+    @Override
+    public String toString() {
+        return "[T]" + super.toString() ;
+    }
+}
```
Good use of inheritance in implementation of toString().

Cheezeblokz commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,97 @@
+                break;
+
+            case "mark":
+                int taskNo = Integer.parseInt(parts[1]);   //assume  saved in part[1], need further improve
```
Comment could be more specific, and dumb downed for the reader instead of for own purpose.

J-Y-Yan committed:
rename a few variables and methods to aviod confusion

J-Y-Yan committed:
create more methods to different classes by code extraction to become OO style

J-Y-Yan committed:
initialise the content of input and expected output txt to conduct a simple output run

bentohset commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,96 @@
+
+            case "deadline":
+                // command e.g. deadline return book /by Sunday
+                String[] ddlSplit = command.split("/");
```
Avoid using short forms as variable names as it might confuse readers with different backgrounds.

bentohset commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,96 @@
+
+        while (true) {
+            String command = keyboard.nextLine();
+            String[] commendSplits = command.split(" ");
```
Small typo on "commendSplits"

bentohset commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    public void setDone(int taskNo, int taskCount, Task[] list){
+        if( (taskNo > taskCount ) || (taskNo <1) ){
+            System.out.println("Oops! You don't have any task in this positions.");
+        }else if(this.isDone){
+            System.out.println("You have already completed the task.");
+        } else{
```
Spacing of the if else statements does not follow standards. Similar in other areas

J-Y-Yan committed:
- add task object creation methods for all kind of task to improve readability of the main(Duke) class
- ready to trackle exception

J-Y-Yan committed:
remove unused variable spaceIndex

J-Y-Yan committed:
- Implement exception to catch some main error caused by user inputs
- Seperate files into different package for easier management

J-Y-Yan committed:
add the delete method and some exception handling for the method

J-Y-Yan committed:
-implement file input ouput methods to sava the tasks data
-add file and data checker methods
       if files or folder not exist, create one;
       if failed to create, throw error for incorrect path
-add backup method to save all currect data in current run of program
-correct bugs on fileIO over creating new task or delete task

J-Y-Yan committed:
create directory 'data' and taskList.txt and corresponding backup file to collect task data

J-Y-Yan committed:
output jar file to release ver. 1.0 application

J-Y-Yan committed:
modify the clearData method so that it only clear data in the beginning when the file is not empty

J-Y-Yan committed:
fixed a bug of inputting consecutive space

J-Y-Yan committed:
add command class to take care of command format issue

J-Y-Yan committed:
introduce command class and make large update to more OO implementation

J-Y-Yan committed:
add time praser functionality

J-Y-Yan committed:
correct bugs on time praser

J-Y-Yan committed:
add find functionality

J-Y-Yan committed:
add more decumentations

J-Y-Yan committed:
add more documentations

J-Y-Yan committed:
launch jar

J-Y-Yan committed:
correct some bugs

J-Y-Yan committed:
correct some issues on running the program over Linux system

J-Y-Yan committed:
Add more documentation

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Bugs fixed in using empty find command and mulfunctions of validating time

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Add more documentations

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Update README.md

J-Y-Yan committed:
Merge pull request #3 from J-Y-Yan/A-JavaDoc2.0

A-JavaDoc
